libs/grok/src/main/java/org/elasticsearch/grok/GrokBuiltinPatterns.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.grok;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class GrokBuiltinPatterns {
+    public static final String[] ECS_COMPATIBILITY_MODES = { "disabled", "v1" };
```

Comment:
I'd love to see `ECS_COMPATIBILITY_MODES` split out into two additional constants -- there's a variety of places where calling code is caring about the position in the array and that bothers me.

That is, I'd like this constant to be `private static final` rather than `public static final`, and all the callers that are using e.g. `ECS_COMPATIBILITY_MODES[1]` would instead use something like `ECS_COMPATIBILITY_V1` instead.

I 100% realize that my issue is with the existing code prior to your change, but since you're here anyway, I'd like to see it fixed.

Reply:
Sure, I can wait.

Something else to put onto your todo list -- it looks to me like `forbidCircularReferences` and its associated helper methods are actually about validating a pattern bank, not a grok, so just fyi that those can probably get moved at some point (but not this PR here now).

Reply:
the same, that's part of the next PRs